acceptancetests/clean_resources.py

```diff
@@ -27,28 +27,62 @@ def get_regions(args, env):
-            logging.info('Cleaning resources in {}.'.format(substrate.region))
-            all_groups = dict(substrate.iter_security_groups())
-            instance_groups = dict(substrate.iter_instance_security_groups())
-            non_instance_groups = dict((k, v) for k, v in all_groups.items()
-                                       if k not in instance_groups)
-            try:
-                unclean = substrate.delete_detached_interfaces(
-                    non_instance_groups.keys())
-                logging.info('Unable to delete {} groups'.format(len(unclean)))
-            except:
-                logging.info('Unable to delete non-instance groups {}'.format(non_instance_groups.keys()))
-            for group_id in unclean:
-                logging.debug('Cannot delete {}'.format(all_groups[group_id]))
-            for group_id in unclean:
-                non_instance_groups.pop(group_id, None)
-            try:
-                substrate.destroy_security_groups(non_instance_groups.values())
-            except:
-                logging.debug('Failed to delete groups {}'.format(non_instance_groups.values()))
+            if substrate is not None:
+                logging.info('Cleaning resources in {}.'
+                             .format(substrate.region))
+                all_groups = dict(substrate.iter_security_groups())
+                instance_groups = dict(substrate.iter_instance_security_groups())
+                logging.info('{} items found from instance groups on {} \n{}'
+                             .format(str(len(instance_groups)),
+                                     region,
+                                     '\n'.join(instance_groups)))
+                non_instance_groups = dict((k, v) for k, v in all_groups.items()
+                                           if k not in instance_groups)
+                logging.info('{} items found from non-instance groups on {} \n{}'
+                             .format(str(len(non_instance_groups)),
+                                     region,
+                                     '\n'.join(non_instance_groups)))
+                try:
+                    logging.info('{} detached interfaces will be deleted from {} \n{}'
+                                 .format(str(len(non_instance_groups.keys())),
+                                         region,
+                                         '\n'.join(non_instance_groups.keys())))
+                    unclean = substrate.delete_detached_interfaces(
+                        non_instance_groups.keys())
+                    logging.info('Unable to delete {} groups from {}'
+                                 .format(str(len(unclean)), region))
+                except:
+                    logging.info('Unable to delete non-instance groups {} from {}'
+                                 .format(non_instance_groups.keys(), region))
+                for group_id in unclean:
```

Comment:
What happens here if delete_detached_interfaces raises an exception above? Unclean will be an unbound variable and will cause an except, stacktrace and exit.